cdap-data-fabric/src/main/java/io/cdap/cdap/spi/data/sql/PostgreSqlStructuredTable.java

```diff
@@ -362,6 +365,25 @@ public CloseableIterator<StructuredRow> scan(Range keyRange, int limit,
   public CloseableIterator<StructuredRow> scan(Range keyRange, int limit,
       Collection<Field<?>> filterIndexes, SortOrder sortOrder)
       throws InvalidFieldException, IOException {
+    return scan(keyRange, limit, filterIndexes, tableSchema.getPrimaryKeys(), sortOrder, false);
+  }
+
+  @Override
+  public CloseableIterator<StructuredRow> scan(Range keyRange, int limit,
+      Collection<Field<?>> filterIndexes, String orderByField, SortOrder sortOrder)
+      throws InvalidFieldException, IOException {
+    if (!tableSchema.isIndexColumn(orderByField) && !tableSchema.isPrimaryKeyColumn(orderByField)) {
+      throw new InvalidFieldException(tableSchema.getTableId(), orderByField,
+          "is not an indexed column or primary key");
+    }
+    return scan(keyRange, limit, filterIndexes, Collections.singleton(orderByField), sortOrder,
+        true);
```

Comment:
It is really confusing that this scan method is an 'and' filter while the other ones are an 'or' filter. Can we just make the operation type part of the primary key? That way it can go in the key range and we can use the existing methods without adding more to StructuredTable. Would like to avoid bloating the StructuredTable class, it's already pretty confusing as-is. 

FYI, whenever we go to CDAP 7.0, we would like to remove Hadoop support, which would allow us to remove the NoSql implementations which would let us use normal SQL instead of this interface.

Reply:
The reason operationtype is not part of the primary key because the operation id and namespace should uniquely identify a operation. This is in line with the API design `/namespace/{namespace}/operation/{id}`. If we add the operationtype in primary key we have to change the API also. 
The API is designed to be compatible with GCP operations api which also does not include types. 
I think we do need a option to select how the filters should be joined. I can create two different top level functions for AND and OR filter.

Reply:
Removed and filter and added operation type in the range key